src/cijoe/core/transport.py

```diff
@@ -35,20 +35,26 @@ def __init__(self, config: Config, output_path: Path):
         self.output_path = output_path
         self.output_ident = "artifacts"
 
-    def run(self, cmd, cwd, env, logfile):
+    def run(self, cmd, cwd, env, logfile, monitor):
         """Invoke the given command"""
 
         env = dict(os.environ)
         env.update(env)
 
         with subprocess.Popen(
             cmd,
-            stdout=logfile,
-            stderr=subprocess.STDOUT,
+            stdout=subprocess.PIPE,
```

Comment:
Don't you need to also redirect stderr to stdout for this to combine the two streams?

Reply:
That depends: do we want `stderr` to be printed as output in the workflow report? With the setup here, it's only printed in the initiator's terminal, but if I set `stderr=subprocess.STDOUT`, both `stderr` and `stdout` will be printed as output in the workflow report.

Reply:
Yup, stderr should be part of output. That is how it is now collected in the various ``run.log`` files.
I was thinking it might be nice to do something like below, inspired by the ``tee`` command.
That is, implement something "file-like" and "with-usable" such that it can be passed to Popen.

```python
class Tee:
    def __init__(self, file_path):
        self.file = open(file_path, 'w')

    def write(self, data):
        self.file.write(data)
        self.file.flush()

        sys.stdout.write(data)
        sys.stdout.flush()

    def flush(self):
        self.file.flush()

    def close(self):
        self.file.close()

    def __enter__(self):
        return self

    def __exit__(self, exc_type, exc_value, traceback):
        self.close()

with Tee('run.log') as tee:
  # Set up the subprocess with stderr redirected to stdout
  process = subprocess.Popen(
    cmd,
    stdout=tee,
    stderr=subprocess.STDOUT,
    ...
  )
```

Reply:
Hmm, might even be added to the state class?

Reply:
Hmm, nevermind, this was a bad suggestion since ``subprocess.Popen()`` does not support "file-like".
The approach of:

```
    stdout=subprocess.PIPE,
    stderr=subprocess.STDOUT,
```

Seems preferable.

Reply:
As we talked about, one problem is that using `subprocess.Popen(..., stdout=tee,...)`, `Popen` will not use the file-like's `write()` method for writing, according to [this StackOverflow thread](https://stackoverflow.com/questions/21035127/how-to-use-a-custom-file-like-object-as-subprocess-stdout-stderr).

I have solved it by modifying the `write()` method of `Tee` and using `PIPE` in `Popen`